server/documents/document.js

```diff
@@ -67,11 +398,58 @@ Meteor.publish('Document.one', function documentOne(args) {
     documentId: Match.DocumentId,
   });
 
+  const self = this;
+  const user = Meteor.user(User.REFERENCE_FIELDS());
+
+  this.autorun((computation) => {
+    if (user) {
+      const fields = Document.PUBLISH_FIELDS();
+      fields.userPermissions = 1;
+      const handle = Document.documents.find(Document.restrictQuery({
+        _id: args.documentId,
+      }, [], user, {$and: [{$or: [{visibility: {$ne: Document.VISIBILITY_LEVELS.PRIVATE}}, {userPermissions: {$elemMatch: {'user._id': user._id, permission: Document.PERMISSIONS.VIEW}}}]}]}), {
+        fields,
+      }).observeChanges({
```

Comment:
You also have to processed `changed`. Otherwise you are now publishing only the document when it is added, but no reactive updates. You have to handle `removed` as well.

(I am working on fixing this, just giving feedback.)